src/main/java/neqsim/thermodynamicoperations/flashops/SolidFlash12.java

```diff
@@ -329,30 +329,30 @@ public void run() {
    * @return a int
    */
   public int checkAndAddSolidPhase() {
-    double[] solidCandidate = new double[system.getPhases()[0].getNumberOfComponents()];
+    double[] solidCandidate = new double[system.getPhase(0).getNumberOfComponents()];
 
     for (int k = 0; k < system.getPhase(0).getNumberOfComponents(); k++) {
       if (system.getTemperature() > system.getPhase(0).getComponent(k)
           .getTriplePointTemperature()) {
         solidCandidate[k] = -10;
       } else {
         solidCandidate[k] = system.getPhase(0).getComponent(k).getz();
-        system.getPhases()[3].getComponent(k).setx(1.0);
+        system.getPhase(3).getComponent(k).setx(1.0);
 
         for (int i = 0; i < system.getNumberOfPhases(); i++) {
-          solidCandidate[k] -= system.getPhases()[3].getComponent(k).fugcoef(system.getPhases()[3])
+          solidCandidate[k] -= system.getPhase(3).getComponent(k).fugcoef(system.getPhase(3))
               / system.getPhase(i).getComponent(k).getFugacityCoefficient();
         }
       }
     }
 
     for (int i = 0; i < solidCandidate.length; i++) {
       if (solidCandidate[i] > 0.0) {
-        system.getPhases()[3].getComponent(i).setx(1.0);
+        system.getPhase(3).getComponent(i).setx(1.0);
         solidIndex = i;
         solidsNumber++;
       } else {
-        system.getPhases()[3].getComponent(i).setx(0.0);
+        system.getPhase(3).getComponent(i).setx(0.0);
```

Comment:
In this case I beleive we have to used system.getPhases()[3] - as the solid phase is always number three (and getPhase(3) can point to a different phase.